datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,362 @@
+            dob = dp_parts[0]
+            h.apply_date(person, "birthDate", dob)
+    if positions:
+        for pos_parts in h.multi_split(positions, [" | ", " / "]):
+            person.add("position", pos_parts)
+    person.add("topics", "poi")
+
+    sanction = h.make_sanction(context, person)
+    sanction.add("reason", data.pop("Reasons").replace(" | ", " "))
+    sanction.add("sourceUrl", link)
+
+    context.emit(person, target=True)
+    context.emit(sanction)
+    context.audit_data(data)
+
+
+def crawl_company(context: Context, link):
+    detail_page = context.fetch_html(link, cache_days=3)
+    details_container = detail_page.find(".//div[@class='col-12 col-lg-9']")
+    data = {}
+    for row in details_container.findall(".//div[@class='row']"):
+        divs = row.findall("div")
+        if len(divs) == 2:
+            label_elem, value_elem = divs
+            if "yellow" in label_elem.get("class"):
+                data = extract_label_value_pair(label_elem, value_elem, data)
+
+    name = data.pop("Name")
+    name_abbr = data.pop("Abbreviated name of the legal entity", None)
+    reg_num = data.pop("Registration number")
+
+    company = context.make("Company")
+    company.id = context.make_id(name, name_abbr, reg_num)
+    company.add("name", name)
+    company.add("name", name_abbr)
+    company.add("registrationNumber", reg_num)
+    company.add("address", data.pop("Address"))
+    company.add("country", data.pop("Country"))
+    company.add("taxNumber", data.pop("Tax Number"))
+    company.add("sourceUrl", data.pop("Links").split(" | "))
+    archive_links = data.pop("Archive links", None)
+    if archive_links is not None:
+        for archive_link in archive_links.split(" | "):
+            company.add("sourceUrl", archive_link)
+
+    company.add("topics", "poi")
+    sanction = h.make_sanction(context, company)
+    sanction.add("reason", data.pop("Reasons").replace(" | ", " "))
+    sanction.add("sourceUrl", link)
+
+    context.emit(company, target=True)
+    context.emit(sanction)
+    context.audit_data(data)
+
+
+def extract_next_page_url(doc, base_url, next_xpath):
+    # next page <a> element extraction using xpath
+    next_link_element = doc.xpath(next_xpath)
+
+    if next_link_element:
+        next_link = next_link_element[0]
+        return next_link.get("href")
+
+    return None
+
+
+def crawl(context: Context):
+    main_page = context.fetch_html(context.data_url, cache_days=3)
+    node = main_page.find(
+        ".//section[@class='sections d-flex flex-wrap align-items-stretch justify-content-center mb-5 pl-5 pr-5 medium']"
+    )
+    h.assert_dom_hash(node, "dbb9a924d940b3a69a132a102e04dcf0f9fbfc5e")
+
+    for link_info in LINKS:
+        base_url = link_info["url"]
+        data_type = link_info["type"]
+        current_url = base_url
+        visited_pages = 0
+        while current_url and visited_pages < 100:  # emergency exit check
```

Comment:
remember to drop this check and rely on `if visited_pages >= 100` at the bottom

```suggestion
        while current_url:
```